src/metaxy/ext/dagster/resource.py

```diff
@@ -0,0 +1,122 @@
+        store = config.get_store(self.store_name)
+
+        if self.fallback_stores is not None:
+            store.fallback_stores = _resolve_fallback_stores(
+                config=config,
+                fallback_names=self.fallback_stores,
+                primary_store=store,
+                primary_store_name=self.store_name or config.store,
+            )
+
+        return store
+
+    @classmethod
+    def from_config(
+        cls,
+        store_name: str | None = None,
+        *,
+        fallback_stores: Sequence[str] | None = None,
+        config_file: str | None = None,
+        search_parents: bool = True,
+        auto_discovery_start: str | None = None,
+    ) -> MetaxyMetadataStoreResource:
+        """Build the resource using Metaxy's configuration."""
+        return cls(
+            store_name=store_name,
+            fallback_stores=list(fallback_stores) if fallback_stores else None,
+            config_file=config_file,
+            search_parents=search_parents,
+            auto_discovery_start=auto_discovery_start,
+        )
+
+    def _load_config(self) -> MetaxyConfig:
+        """Load the Metaxy configuration respecting explicit overrides."""
+        config_file = Path(self.config_file) if self.config_file is not None else None
+        auto_discovery_start = (
+            Path(self.auto_discovery_start)
+            if self.auto_discovery_start is not None
+            else None
+        )
+
+        if config_file is not None:
+            return MetaxyConfig.load(
+                config_file=config_file,
+                search_parents=self.search_parents,
+                auto_discovery_start=auto_discovery_start,
+            )
+
+        if MetaxyConfig.is_set():
+            return MetaxyConfig.get()
+
+        return MetaxyConfig.load(
+            search_parents=self.search_parents,
+            auto_discovery_start=auto_discovery_start,
+        )
+
+
+def _resolve_fallback_stores(
+    *,
+    config: MetaxyConfig,
+    fallback_names: Sequence[str],
+    primary_store: MetadataStore,
+    primary_store_name: str,
+) -> list[MetadataStore]:
+    """Instantiate fallback stores from names and validate compatibility."""
+    seen: set[str] = set()
+    fallbacks: list[MetadataStore] = []
+
+    for name in fallback_names:
+        if name in seen:
+            raise ValueError(f"Duplicate fallback store '{name}' in configuration.")
+        if name == primary_store_name:
+            raise ValueError("A store cannot list itself as a fallback store.")
+
+        fallback_store = config.get_store(name)
+        _validate_fallback_store(primary_store, fallback_store, name)
+
+        fallbacks.append(fallback_store)
+        seen.add(name)
+
+    return fallbacks
+
+
+def _validate_fallback_store(
+    primary_store: MetadataStore, fallback_store: MetadataStore, fallback_name: str
+) -> None:
+    """Ensure fallback stores share hashing settings with the primary store."""
```

Comment:
I think this would make a lot of sense, but is not stringently an absolute necessity. But should we endorse it (mixing hashes)? I think the more standard/simple way forward is actually for not mixing - and for not allowing it here